### PR TITLE
[PLSQL] Single statement without ending semicolon

### PR DESCRIPTION
**Pull request by mfvitale**
In case of single statement without ending semicolon, do you think this should pass?

I think yes, since this query is valid. The statement should fail if is followed by another one.

### CODE REVIEW

**Comment by mfvitale** [comment]
@KvanTTT To let single statement working without semicolon required lot of changes. Can I gently ask to a review?

**Comment by mfvitale** [comment]
@KvanTTT Can you please help with `Parsing grammar /home/runner/work/grammars-v4/grammars-v4/sql/plsql/Generated-CSharp/PlSqlParser.g4
2023-06-30T09:10:52.9763475Z Process is taking longer than 300 seconds. Killing process.
2023-06-30T09:10:53.0751479Z Test failed.
2023-06-30T09:10:53.0776720Z  Test 00:08:18`

I need some guidance on how to figure out the issue here. Thanks.

File: sql/plsql/PlSqlLexer.g4 [hunk 1]
```diff
@@ -1655,6 +1655,7 @@ SQLDATA:                      'SQLDATA';
 SQLERROR:                     'SQLERROR';
 SQLLDR:                       'SQLLDR';
 SQL:                          'SQL';
+FILE_EXT:                     ('PKB' | 'PKS');
```
Comment (KvanTTT) [inline sql/plsql/PlSqlLexer.g4:1658]:
```suggestion
FILE_EXT:                     'PKB' | 'PKS';
```

File: sql/plsql/PlSqlParser.g4 [hunk 2]
```diff
@@ -30,7 +30,7 @@ options {
 }
 
 sql_script
-    : ((sql_plus_command | unit_statement) SEMICOLON?)* EOF
+    : sql_plus_command_no_semicolon? (((sql_plus_command | unit_statement) (SEMICOLON'/'? (sql_plus_command | unit_statement))* SEMICOLON?'/'?)) EOF
```
Comment (KvanTTT) [inline sql/plsql/PlSqlParser.g4:33]: Please always put space after terms and remove useless parentheses:

```suggestion
    : sql_plus_command_no_semicolon? ((sql_plus_command | unit_statement) (SEMICOLON '/'? (sql_plus_command | unit_statement))* SEMICOLON? '/'?) EOF
```

**Review by KvanTTT — CHANGES_REQUESTED** [review]

**Comment by mfvitale** [comment]
@KvanTTT  thanks for your suggestions. I have applied it.

**Review by KvanTTT — APPROVED** [review]

**Comment by KvanTTT** [comment]
@kaby76 it looks like there is a problem with coverage CI task. Could you investigate please?

**Comment by mfvitale** [comment]
> @kaby76 it looks like there is a problem with coverage CI task. Could you investigate please?

Seems same issue with CSharp.

**Comment by kaby76** [comment]
I mentioned this for another PR. https://github.com/antlr/grammars-v4/pull/3612#issuecomment-1636193153

And I opened an Issue for it. https://github.com/antlr/grammars-v4/issues/3597

I will turn off voverage computations for now as this will require a change to trcover, and will take time to make sure it works.

**Comment by Thracky** [comment]
I should probably open an issue for this but SQLPlus commands do not require a terminating semicolon at all. See section 4.6.1 of the SQLPlus user guide here: https://docs.oracle.com/en/database/oracle/oracle-database/23/sqpug/SQL-Plus-basics.html#GUID-4F029548-6DD2-468F-B6E1-E6D6AA59014B

So if you have a file that starts with:

```
WHENEVER SQLERROR EXIT

SET ECHO ON

CREATE OR REPLACE PACKAGE ...
```

These changes make the parser bail out immediately on the `SET` and it is perfectly legal syntax for SQLPlus commands.

**Pull request merged**